docs/decisions/0004-downstream-effect-events.rst

```diff
@@ -0,0 +1,103 @@
+  Instructor Mark Subsection Completed   When an exam attempt is completed.                                                               username, content_id                                                                        lms/djangoapps/instructor/signals.py         update_exam_completion_task in tasks.py         When we need to mark a subsection as completed.                           A subsection is marked completed for a learner.                                       
+  Invalidate Certificate                 When an exam attempt is rejected.                                                                user_id, course_key_or_id                                                                   lms/djangoapps/certificates/signals.py       invalidate_certificate in services.py           When we need to invalidate a learner's certificate.                       A certificate object's status is set to "unavailable".                                
+  Set Credit Requirement Status          When exam attempt is completed.                                                                  user_id, course_key_or_id, req_namespace, req_name, status                                  openedx/core/djangoapps/credits/signals.py   set_credit_requirement_status in services.py    When we need to create or modify a learner's credit requirement status.   A credit requirement status object is created or modified within the LMS.             
+ ====================================== ================================================================================================ =========================================================================================== ============================================ =============================================== ========================================================================= ====================================================================================== 
+
+Consequences
+------------
+#. Event definitions implemented in openedx-events
+
+  * Defining the events and the data sent in each in this abstraction layer is fundamental to making event bus work.
+
+  * We have designed these events to be "generic", such that they can be triggered under contexts outside of exams by other services.
+
+#. Event producers implemented to edx-exams
+
+  * We will implement these producers in the backend in the places we want these events to be triggered.
+
+#. Event consumers added to edx-platform
+
+  * We will add consumers in the signals.py file in each edx-platform service's respective folders.
+
+  * These consumers will call other service or api functions in those folders.
+
+References
+----------
+
+* Discovery Doc for M6: https://2u-internal.atlassian.net/wiki/spaces/PT/pages/539066520/MST-1789+M6+Exam+Review+and+Downstream+Triggers+Scope+Definition+and+Story+Writing#Implementation:
```

Comment:
This document is not going to be visible to the Open edX community because it's in 2U's internal Confluence instance. We should either not link to it or consider moving this to the Open edX Confluence space.

Reply:
I'm a bit confused by this, as I do see some links to the internal confluence in the other ADR files within this folder, e.g. https://github.com/edx/edx-exams/blob/main/docs/decisions/0002-multiple-exam-systems.rst?plain=1#L54

Reply:
Yeah, I think we have been inconsistent with this. I don't think the feedback here is based on opinions you don't have to adopt. You can do whatever you feel is easy.